projects/control-service/projects/pipelines_control_service/src/integration-test/java/com/vmware/taurus/datajobs/it/DataJobDeploymentCrudAsyncIT.java

```diff
@@ -24,6 +29,61 @@
     webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT,
     classes = ControlplaneApplication.class)
 public class DataJobDeploymentCrudAsyncIT extends BaseDataJobDeploymentCrudIT {
+
+  @Autowired private DesiredJobDeploymentRepository desiredJobDeploymentRepository;
+  @Autowired private ActualJobDeploymentRepository actualJobDeploymentRepository;
+
+  @Override
+  protected void beforeDeploymentDeletion() {
+    var desiredDataJobDeployment = desiredJobDeploymentRepository.findById(testJobName);
```

Comment:
why don't you use existsById here and then you can do the assertions here, removing line 41 and 42.